File: firmware/test_neopixel.py
"""Stage 3 roadmap task 18: test the NeoPixel (WS2812) color LEDs independently.

Run on-device and record the observed colors as evidence in
docs/PROJECT_STATE.md before marking this task done.
"""
from machine import Pin
from neopixel import NeoPixel
import time

rgb = NeoPixel(Pin(10, Pin.OUT), 3)

for color in ((50, 0, 0), (0, 50, 0), (0, 0, 50)):  # red, green, blue
    for i in range(3):
        rgb[i] = color
    rgb.write()
    time.sleep_ms(1000)

rgb.fill((0, 0, 0))
rgb.write()

File: firmware/src/test_neopixel.cpp
// Stage 3 roadmap task 18: test the onboard NeoPixel (WS2812) LEDs
// independently (env:test_neopixel).
//
// Build/upload with: pio run -e test_neopixel -t upload
// Record the observed colors as evidence in docs/PROJECT_STATE.md before
// marking this task done.
#include <Arduino.h>
#include <Adafruit_NeoPixel.h>

#include "pins.h"

Adafruit_NeoPixel rgb(pins::NEOPIXEL_COUNT, pins::NEOPIXEL, NEO_GRB + NEO_KHZ800);

void setup() {
  rgb.begin();
}

void loop() {
  const uint32_t colors[] = {
      rgb.Color(50, 0, 0),  // red
      rgb.Color(0, 50, 0),  // green
      rgb.Color(0, 0, 50),  // blue
  };
  for (uint32_t color : colors) {
    for (int i = 0; i < pins::NEOPIXEL_COUNT; i++) {
      rgb.setPixelColor(i, color);
    }
    rgb.show();
    delay(1000);
  }
}
